test: Add tests for the add and remove item features

gvgesanto2 opened the pull request:
## Description

**Task:** https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md

This PR makes the following changes:
- Adds tests for the TasksList.createAndAddTask() and View.render() methods (add item feature)
- Adds tests for the TasksList.removeTaskById() and View.remove() methods (remove item feature)
- Adds tests for the StorageService class

Plus, it adopts the following methodologies and conventions:
- For the CSS/Sass:
  - [BEM](https://en.bem.info/methodology/) methodology for naming the CSS classes
  - A modified version of the [Scalable and Modular Architecture for CSS](http://smacss.com/) (SMACSS) methodology for organizing the CSS code into categorized folders/files (base, components, layout, and utilities)
  - [Atomic Design](https://bradfrost.com/blog/post/atomic-web-design/) methodology for breaking down the CSS components into atoms, molecules, and organisms
- For the JavaScript:
  - [MVP architecture](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter) (a variation of the [MVC pattern](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller)) to separate the application business logic from the presentation logic (DOM manipulation tasks)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Tests (changes related to testing)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented my code
- [X] I have deleted any unused code as comments in my codebase

## Screenshots

![add-remove-tests](https://user-images.githubusercontent.com/64566209/172937942-4a8d76d5-782d-4f48-86c8-51ae455649e7.png)

gvgesanto2 committed:
test: Add tests for the TasksList and StorageService classes

gvgesanto2 committed:
test: Test the render HTML element feature

chumaachike committed:
view.remove and view.append tests added

gvgesanto2 committed:
fix: Fix linters errors

medaminedev66 reviewed (changes requested):
Hi @chumaachike and @gvgesanto2  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional ✔️
- There are no linter errors ✔️
- You have a descriptive PR ✔️
- Your tests run succussfully✔️

## Required Changes ♻️
- [ ] It would be preferable if you provide instructions in your README on how users can run the tests locally.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gvgesanto2 committed:
docs: Update README

gvgesanto2 commented:
## Description

This PR makes the following changes:
- Updates the README file with the instructions on how users can run the tests locally as requested in the [last review](https://github.com/gvgesanto2/microverse-todo-list/pull/7#pullrequestreview-1002003311)

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented my code
- [X] I have deleted any unused code as comments in my codebase

mikethreels reviewed (approved):
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
